instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/instrumentation.rb

```diff
@@ -51,7 +51,19 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         def install_middleware(app)
-          app.use(*OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.middleware_args) if config[:install_rack]
+          stability_opt_in = ENV.fetch('OTEL_SEMCONV_STABILITY_OPT_IN', '')
+          values = stability_opt_in.split(',').map(&:strip)
+
+          if config[:install_rack]
+            if values.include?('http/dup')
+              app.use(*OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.middleware_args_dup)
+            elsif values.include?('http')
+              app.use(*OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.middleware_args_stable)
+            else
+              app.use(*OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.middleware_args_old)
+            end
+          end
+
```

Comment:
Sinatra instrumentation relies on installing Rack middleware.

Reply:
Added tests for this too https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1594/commits/6dca891bf0b820b58a493d966e9fefb3c80fedce